script/main.py

```diff
@@ -22,27 +22,53 @@ def init_argparser():
   return args
 
 
-def find_record_with_dob(people_json):
-  record_names = []
+def find_records(people_json):
+  record_names_phone = []
+  record_names_with_dob =[]
   dict_json_to_list = people_json['results']
   for dict_single_record in dict_json_to_list:
+    if 'phone' and 'cell' in dict_single_record:
+      record_names_phone.append(dict_single_record)
     if 'dob' in dict_single_record:
-      record_names.append(dict_single_record)
-  return record_names
+      record_names_with_dob.append(dict_single_record)
+  return (record_names_phone, record_names_with_dob)
 
 
-def create_new_record_with_dob_in_json(records):
-  for record in records:
+def is_leap_year(year):
+  return year % 4 == 0 and (year % 100 != 0 or year % 400 == 0)
+
+
+def create_new_record_with_dob_in_json(dob_records):
+  for record in dob_records:
     born = record['dob']['date']
     full_date_of_birth = datetime.strptime(born, '%Y-%m-%dT%H:%M:%S.%fZ')
     date_of_birth = full_date_of_birth.date()
     today = date.today()
+    is_leap_year_birthday = False
+    if date_of_birth.month == 2 and date_of_birth.day == 29 and is_leap_year(today.year) == False:
+      is_leap_year_birthday = True
+      date_of_birth = date_of_birth.replace(day=28)
     date_of_the_nearest_birthday = date_of_birth.replace(year=today.year)
     if date_of_the_nearest_birthday < today:
+      if date_of_birth.month == 2 and date_of_birth.day == 29 and is_leap_year(today.year + 1) == False:
+        date_of_birth = date_of_birth.replace(day=28)
       date_of_the_nearest_birthday = date_of_birth.replace(year=today.year +1)
-    days_left = abs(date_of_the_nearest_birthday - today).days
-    dob_new_record_time_until_birthday = {"time_until_birthday": str(days_left)}
+      if is_leap_year_birthday:
+        date_of_the_nearest_birthday = date_of_the_nearest_birthday.replace(day=29)      
+    days_left = abs(date_of_the_nearest_birthday - today).days    
+    dob_new_record_time_until_birthday = {"time_until_birthday": days_left}
     record["dob"].update(dob_new_record_time_until_birthday)
+  
+
+def removes_special_characters_from_phone_and_cell_numbers(phone_records):
+  for record in phone_records:
+    phone = record['phone']       
+    clear_phone = phone.replace("(", "").replace(")", "").replace("-", "").replace("+", "").replace(" ", "")
```

Comment:
You can use regex here

Reply:
I don't know how to use "str.replace ()" and "regex" together

Reply:
check:
https://stackoverflow.com/questions/11475885/python-replace-regex

Reply:
I've seen it before, but it doesn't mean anything to me. Here they recommended using "re.sub", which I do in my code.

Reply:
When I changed the code to this:
```
  phone = record['phone']       
    clear_phone = phone.replace('\(|\)|\-|\+|\s', '')
    record['phone'] = clear_phone

    cell = record['cell'] 
    clear_cell = cell.replace('\(|\)|\-|\+|\s', '')
    record['cell'] = clear_cell
``` 
Nothing happens :\

Reply:
You are not using re.sub here.
Check the code:
```
re.sub(r'a', 'b', 'banana')
```
How can you modify it to get what you want?
Also regarding the regex itself. You are only interested in digits. How to write regex to replace all not digit characters?

Reply:
In my last changes I have this:

```
phone = record['phone']       
    clear_phone = re.sub(r'\(|\)|\-|\+|\s', '', phone)
    record['phone'] = clear_phone

    cell = record['cell'] 
    clear_cell = re.sub(r'\(|\)|\-|\+|\s', '', cell)
    record['cell'] = clear_cell
```

Reply:
Ok. this is correct. But you can improve it in the following way
You want only numbers from the phone number, yes? So to match any character which is not number you can use the following regex expression
```
clear_phone = re.sub(r'\D', '', phone)
```